sevntu-checks/pom.xml

```diff
@@ -283,7 +294,9 @@
               <regex><pattern>.*.checks.coding.MapIterationInForEachLoopCheck</pattern><branchRate>90</branchRate><lineRate>98</lineRate></regex>
               <regex><pattern>.*.checks.coding.NoNullForCollectionReturnCheck</pattern><branchRate>85</branchRate><lineRate>96</lineRate></regex>
               <regex><pattern>.*.checks.coding.OverridableMethodInConstructorCheck</pattern><branchRate>94</branchRate><lineRate>99</lineRate></regex>
-              <regex><pattern>.*.checks.design.HideUtilityClassConstructorCheck</pattern><branchRate>94</branchRate><lineRate>100</lineRate></regex>
+              <regex><pattern>.*.checks.coding.Jsr305AnnotationsCheck</pattern><branchRate>96</branchRate><lineRate>98</lineRate></regex>
+              <regex><pattern>.*.checks.coding.Jsr305AnnotationsCheck.AbstractJsr305Check</pattern><branchRate>98</branchRate><lineRate>100</lineRate></regex>
+              <regex><pattern>.*.checks.design.HideUtilityClassConstructorCheck</pattern><branchRate>96</branchRate><lineRate>100</lineRate></regex>
```

Comment:
This is not good, I hope you still working on coverage.

Reply:
Please take a look at the [jacoco report](https://user-images.githubusercontent.com/383271/56152594-6c76fc80-5fb4-11e9-8668-c508f25ded6b.jpg) (it is a zip file regardless the extension). 

The coverage is actually pracitally 100%. There is only one missed branch: 

![screenshot-2019-04-17-13:28:58](https://user-images.githubusercontent.com/383271/56284477-d4dcef80-6114-11e9-83f0-32dbac11a42a.png)

When run on my box, `mvn cobertura:check` reports the following:
1. `Jsr305AnnotationsCheck$AbstractJsr305Check`: branch coverage 99.7%, line coverage 100%
2. `Jsr305AnnotationsCheck`: branch coverage 98.1%, line coverage 100%

For no obvious reason the same check on travis reports lower coverage, hence I set even lower limits in the _pom.xml_. 

Also on my machine cobertura complains about a line coverage rate of 87.5% in `NumericLiteralNeedsUnderscoreCheck$NumericType` which seems to be 100% when executed on travis. 

I am using the same commands as found in `.ci/travis.sh`.

Reply:
Such default, is ok cover by direct execution of method with bad parameters. We usually do this by reflection, as we discourage use of protected scope.

Reply:
Sorry, I don't quite understand.

Reply:
We need to get 100% coverage, your case is exceptional case that must not happen, but exist to give proper signal of something missed to next engineer who update Check. The only way to get execution of such code in switch-default is by reflection. You have to reach 100% code coverage.

Reply:
It is better to share reports as link to GitHub.io , to let us review it and reply to your sooner ( from phone, on the go)

Reply:
> It is better to share reports as link to GitHub.io , to let us review it and reply to your sooner ( from phone, on the go)

Makes sense. Here it is: https://mbert.github.io/sevntu-checkstyle/coverage/jacoco/index.html

> We need to get 100% coverage, your case is exceptional case that must not happen, but exist to give proper signal of something missed to next engineer who update Check. The only way to get execution of such code in switch-default is by reflection. You have to reach 100% code coverage.

As I learned from @rnveach, tests that don't consist of running the check against input files should not be used. I can't think of any method to cover this code line while sticking to this rule.

Having checked how other checks handle this (it would usually be found in `visitToken()` I found that a good number of them use similar `switch` constructions - and don't cover such defaults either. 

Any idea  how to proceed here?

Reply:
yes, we are not born smart, it took us some time to understand some best practices.
We are more fanatic in main project, where all technical debt(debt of 16 years) is resolved.
See example - https://github.com/checkstyle/checkstyle/blob/a73ff0890c27cbe8affc1a55f2710a231687bd85/src/test/java/com/puppycrawl/tools/checkstyle/checks/design/ThrowsCountCheckTest.java#L85 and https://github.com/checkstyle/checkstyle/blob/9a96cc860d7f1394ed8d1a60b7b58e0ea3613128/src/test/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheckTest.java#L273, but please put a brief comment above test method to explain why we do hack in testing.
You can even access 'handleDefinition' method by reflection, it is allowed.

Reply:
Done.

Reply:
Also squashed commits as mentioned by @rnveach